### PR TITLE
Update the Windows installer to better handle gcc toolchain in differ…

### PR DESCRIPTION
**Pull request by Tieske**
…ent environments

This change will hardcode the toolchain utilities

- after this change it will require the toolchain to be in the path when installing LR
- manually tested with MinGW (old), MinGW-w64 (actively maintained), and both 32&64 bit TDM releases
- effect on MS toolchain untested

This should make LR compatible with modern gcc versions.

example config file:
```lua
rocks_trees = {
    { name = [[user]],
         root    = home..[[/luarocks]],
    },
    { name = [[system]],
         root    = [[c:\lua\]],
    },
}
variables = {
    MSVCRT = 'm',   -- make MinGW use MSVCRT.DLL as runtime
    LUALIB = 'lua51.dll',
    CC = [[C:\mingw-w64\i686-7.2.0-posix-dwarf-rt_v5-rev1\mingw32\bin\gcc.exe]],
    MAKE = [[C:\mingw-w64\i686-7.2.0-posix-dwarf-rt_v5-rev1\mingw32\bin\mingw32-make.exe]],
    RC = [[C:\mingw-w64\i686-7.2.0-posix-dwarf-rt_v5-rev1\mingw32\bin\windres.exe]],
    LD = [[C:\mingw-w64\i686-7.2.0-posix-dwarf-rt_v5-rev1\mingw32\bin\gcc.exe]],
    AR = [[C:\mingw-w64\i686-7.2.0-posix-dwarf-rt_v5-rev1\mingw32\bin\ar.exe]],
    RANLIB = [[C:\mingw-w64\i686-7.2.0-posix-dwarf-rt_v5-rev1\mingw32\bin\ranlib.exe]],
}
verbose = false   -- set to 'true' to enable verbose output
```

### CODE REVIEW

File: install.bat [hunk 1]
```diff
@@ -617,6 +624,66 @@ local function restore_config_files()
...
+            result[name] = find_file("*"..name.."*.exe", path)
+        end
+    end
+
+    vars.MINGW_MAKE = (result.make and '[['..result.make..']]') or "nil  -- not found by intaller"
+    vars.MINGW_CC = (result.gcc and '[['..result.gcc..']]') or "nil  -- not found by intaller"
+    vars.MINGW_RC = (result.windres and '[['..result.windres..']]') or "nil  -- not found by intaller"
+    vars.MINGW_LD = (result.gcc and '[['..result.gcc..']]') or "nil  -- not found by intaller"
+    vars.MINGW_AR = (result.ar and '[['..result.ar..']]') or "nil  -- not found by intaller"
+    vars.MINGW_RANLIB = (result.ranlib and '[['..result.ranlib..']]') or "nil  -- not found by intaller"
```
Comment (hishamhm) [inline install.bat]: typo in all entries above: `intaller` -> `installer`

File: install.bat [hunk 2]
```diff
@@ -617,6 +624,66 @@ local function restore_config_files()
...
+        end
+    end
+    
+    -- returns: full filename, path, filename
+    local find_file = function(mask, path)
+        local cmd
+        if path then
+            cmd = 'where.exe /R "' .. path .. '" ' .. mask
+        else
+            cmd = 'where.exe ' .. mask
```
Comment (hishamhm) [inline install.bat]: I assume `where.exe` is a standard Windows command? (it's not included in the LuaRocks repo or this PR)

Comment (Tieske) [reply install.bat]: yes, from Windows Server 2003 onwards (so not on Win XP)

Comment (Tieske) [reply install.bat]: its similar to unix `which`

File: install.bat (hunk 1 above)
Comment (Tieske) [reply install.bat]: thx. Fixed.

**Comment by hishamhm** [comment]
@Tieske thanks! I'm trying to fix the Appveyor CI now so we can run this PR

**Pull request merged**